subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/AbstractJavaGroovyIncrementalCompilationSupport.groovy

```diff
@@ -29,7 +29,7 @@ abstract class AbstractJavaGroovyIncrementalCompilationSupport extends AbstractI
     File source(String... classBodies) {
         File out
         for (String body : classBodies) {
-            def className = (body =~ /(?s).*?(?:class|interface|enum) (\w+) .*/)[0][1]
+            def className = (body =~ /(?s).*?(?:class|interface|enum) ([\w$]+) .*/)[0][1]
```

Comment:
Are there any other characters we are missing here I wonder?

Reply:
This is test code, so it doesn't really matter, right?

Reply:
It seems a lot of characters can be in Java class name. For example this compiles:
```
public class π {
}
```
:D 

So yeah, I would add them to this regex when needed.